### ISSUE TITLE
Can MySQL module change labels to MariaDB when that database is used?

### ISSUE THREAD

**Issue by swelljoe** [webmin/webmin#1421]
Jamie, would it be feasible/reasonable and not messy code-wise to make the MySQL module display `MariaDB` instead of `MySQL`, just to eliminate confusion among users about which database is installed and in use?

They're close enough to interchangeable for our purposes most of the time, but we still get folks who are confused about what database they have, sometimes. (Though maybe changing the label will make more people think they need to "upgrade" to MySQL...folks will always find a way to make their lives harder.)

I don't see an obvious way to do it, since it's all generated with variables from the text hash, but you're much more clever than I, so maybe you see an easy/clean way to do it.

**Comment by jcameron** [comment]
For the text within the MySQL module, it wouldn't be too hard to do this. The only potentially hard part would be the module title on the left menu.

Are you also thinking we should do this in the Virtualmin UI?

**Comment by swelljoe** [comment]
Everywhere that MySQL currently appears on a system that actually has MariaDB. I wouldn't want you to spend a lot of time on it, though. It's a minor thing that I think documentation can also address. It's just been a source of occasional confusion...but, I think it's fine to kick it down the road for later, if it's at all difficult.

I think some of our users don't really understand that MySQL and MariaDB are effectively interchangeable. and having the label say MySQL when it's really MariaDB confuses some of them. (But, then again, those same users might be confused into trying to install MySQL if they see they "only" have "MariaDB" and know that their apps need "MySQL".)

This may just not be a problem with a technical solution, I'm not sure! It feels righter to call it what it is, but it probably won't actually solve the problem of user confusion.

**Comment by iliarostovtsev** [comment]
> For the text within the MySQL module, it wouldn't be too hard to do this. 

Yes, how about [this](https://github.com/webmin/webmin/commit/371be571af50c45b1cbe654d239d3aaf2ae9f39e)? 

.. but we also have _config_, _module.config_ and _help_ to deal with. It can be also pretty easily doable but I am not sure about _not messy code-wise_, as Joe mentioned before. Jamie, how about modifying `get_module_info()` and `help_file` subs?

**Comment by jcameron** [comment]
@iliaross yes something like that, however we'd want to aggressively cache the variant of MySQL in use so that we don't have to hit the database everytime the `mysql` module is loaded.

Ironically, I often see people confused in the other direction - they still refer to "MySQL" when it's actually MariaDB!

**Comment by iliarostovtsev** [comment]
@jcameron Yeah, caching is not a problem. What about `get_module_info()` and `help_file` subs? Is this how we should handle a left menu and and help replacements?

**Comment by iliarostovtsev** [comment]
(by not a problem, I meant, I'd add it)

**Comment by jcameron** [comment]
I don't have a good solution for get_module_info, sorry .. I'll have to think about how it can be done safely.

**Comment by jcameron** [comment]
I've also updated Virtualmin to also show "MariaDB" in it's part of the UI if that variant is installed.

**Comment by iliarostovtsev** [comment]
So, the left menu link for Webmin is the last obstacle?

If we cache db server variant to `$module_config/version`, perhaps we could quickly read this file when building left menu links. The only question how to do it elegantly code-wise?

**Comment by jcameron** [comment]
Yeah, that could work - the only hard part is figuring out how to do it cleanly.

**Comment by iliarostovtsev** [comment]
Jamie, how about this for overriding menu entry for MySQL - https://github.com/webmin/webmin/pull/1470/files
